code/modules/antagonists/eldritch_cult/eldritch_effects.dm

```diff
@@ -0,0 +1,288 @@
+	var/datum/antagonist/heretic/cultie = user.mind.has_antag_datum(/datum/antagonist/heretic)
+	var/list/knowledge = cultie.get_all_knowledge()
+	
+	var/list/name_list = list()
+
+	is_in_use = TRUE
+	for(var/X in knowledge)
+		var/datum/eldritch_knowledge/current_eldritch_knowledge = knowledge[X]
+		if(codex.charge - current_eldritch_knowledge.charges_needed < 0 || (current_eldritch_knowledge.charges_needed == 0 && current_eldritch_knowledge.force_showup == FALSE) || !current_eldritch_knowledge.recipe_snowflake_check(user,buckled_human,loc))
+			continue
+		name_list += current_eldritch_knowledge.name
+		name_list[current_eldritch_knowledge.name] = current_eldritch_knowledge
+	
+	var/chosen = input("Select the ritual you wish to perform","Your target") as null|anything in sortList(name_list, /proc/cmp_text_asc)	
+	if(!chosen)
+		return
+	flick("[icon_state]_active",src)
+	playsound(user, 'sound/magic/castsummon.ogg', 75, TRUE)
+		//we are doing this since some on_finished_recipe subtract the atoms from selected_atoms making them invisible permanently.
+	var/datum/eldritch_knowledge/chosen_knowledge = name_list[chosen]
+	codex.charge -= chosen_knowledge.charges_needed
+	if(!chosen_knowledge.on_finished_recipe(user,buckled_human,loc))
+		to_chat(user,"<span class='warning'>Your ritual failed! You used either wrong components or are missing something important!</span>")
+	
+	is_in_use = FALSE
+	
+/obj/effect/eldritch/big
+	name = "Transmutation rune"
+	icon = 'icons/effects/96x96.dmi'
+	icon_state = "eldritch_rune1"
+	pixel_x = -32 //So the big ol' 96x96 sprite shows up right
+	pixel_y = -32
+
+/**
+  * #Reality smash tracker
+  *
+  * Stupid fucking list holder, DONT create new ones, it will break the game, this is automnatically created whenever eldritch cultists are created.
+  *
+  * Tracks relevant data, generates relevant data, useful tool
+  */
+/datum/reality_smash_tracker
+	///list of tracked reality smashes
+	var/list/smashes = list()
+	///List of mobs with ability to see the smashes
+	var/list/targets = list()
+
+/datum/reality_smash_tracker/New()
+	. = ..()
+	addtimer(CALLBACK(src, .proc/Regenerate), 20 MINUTES)
+
+/datum/reality_smash_tracker/Destroy(force, ...)
+	if(GLOB.reality_smash_track == src)
+		stack_trace("/datum/reality_smash_tracker was deleted. Heretics may no longer access any influences. Fix it or call coder support")
+	QDEL_LIST(smashes)
+	targets.Cut()
+	return ..()
+
+/datum/reality_smash_tracker/proc/Regenerate()
+	//we can skip the istype checks on for loop by adding as anything.
+	for(var/mob/mobie as anything in targets)
+		to_chat(mobie, "<span class='notice'>You feel reality shift, changing the influences...</span>")
+	addtimer(CALLBACK(src, .proc/Regenerate), 20 MINUTES)
+	QDEL_LIST(smashes)
+	_Generate()
+
+/**
+  * Automatically fixes the target and smash network
+  *
+  * Fixes any bugs that are caused by late Generate() or exchanging clients
+  */
+/datum/reality_smash_tracker/proc/ReworkNetwork()
```

Comment:
Is this common?

Reply:
happens every 20 minutes when influences regenerate or someone reconnectects. It was either dealing with this OR shitcoding invisibility code.